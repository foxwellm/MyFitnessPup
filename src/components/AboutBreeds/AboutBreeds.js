import React, { Component } from 'react'
import { connect } from 'react-redux'
import BreedCard from '../BreedCard/BreedCard'
import { Link } from 'react-router-dom'
let shortID = require('short-id');

export class AboutBreeds extends Component {

  render() {
    const { staticBreeds, staticBreedInfo, location } = this.props
    const breedCards = staticBreeds.map((breed, i) => {
      return <Link to={`/about-breeds/${breed.breed}`} key={shortID.generate()}>
        <BreedCard
          {...breed}
          info={staticBreedInfo[breed.name]}
          location={location}
          active={false}
          cardNumber={i}
        />
      </Link>
    })

    return (
      <div className='breed-cards-container'>{breedCards}</div >
    )
  }
}

export const mapStateToProps = (state) => ({
  staticBreeds: state.staticBreeds,
  staticBreedInfo: state.staticBreedInfo,
})

export default connect(mapStateToProps)(AboutBreeds)