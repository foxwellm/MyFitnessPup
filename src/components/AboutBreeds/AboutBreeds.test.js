import React from 'react'
import { shallow } from 'enzyme'
import AboutBreeds from './AboutBreeds'

describe('AboutBreeds', () => {
  let wrapper
  it('should have match the correct snapshot', () => {
    let mockStaticBreeds = [{ breed: 'Husky' }, { breed: 'Chihuahua' }]
    let mockStaticBreedInfo = [{ name: 'Husky' }, { name: 'Chihuahua' }]
    let mockLocation = '77043'
    wrapper = shallow(<AboutBreeds
      staticBreeds={mockStaticBreeds}
      staticBreedInfo={mockStaticBreedInfo}
      location={mockLocation}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})