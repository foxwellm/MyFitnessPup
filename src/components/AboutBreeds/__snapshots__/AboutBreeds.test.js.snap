// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutBreeds should have match the correct snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
